doc/cephfs/charmap.rst

```diff
@@ -0,0 +1,185 @@
+    # file: foo/
+    ceph.dir.charmap="{\"casesensitive\":true,\"normalization\":\"nfd\",\"encoding\":\"utf8\"}"
+    
+    $ getfattr -n ceph.dir.normalization foo/
+    # file: foo/
+    ceph.dir.normalization="nfd"
+
+.. note:: Setting the empty string will cause the MDS to pick the default normalization.
+
+All character mapping configurations must have a normalization enabled. Removing the normalization
+will cause the default to be restored:
+
+::
+
+    $ setfattr -n ceph.dir.normalization -v nfc foo/
+    $ getfattr -n ceph.dir.normalization foo/
+    # file: foo/
+    ceph.dir.normalization="nfc"
+    
+    $ setfattr -x ceph.dir.normalization foo/
+    $ getfattr -n ceph.dir.normalization foo/
+    # file: foo/
+    ceph.dir.normalization="nfd"
+
+To remove normlization on a directory, you must remove the ``ceph.dir.charmap``
+configuration.
+
+.. note:: The MDS maintains an ``alternate_name`` metadata (also used for
+          encryption) for directory entries which allows the client to persist the
+          original un-normalized name used by the application. The MDS does not
+          interpret this metadata in any way; it's only used by clients to reconstruct
+          the original name of the directory entry.
+
+
+Case Folding
+------------
+
+The ``ceph.dir.casesensitive`` configuration accepts a boolean value. By
+default, names are case-sensitive (as normal in a POSIX file system). Setting
+this value to false will make the directory (and its children)
+case-insensitive.
+
+Case folding requires that names are also normalized. By default, after setting
+a directory to be case-insensitive, the **charmap** will be:
+
+::
+
+    $ setfattr -n ceph.dir.casesensitive -v 0 foo/
+    $ getfattr -n ceph.dir.casesensitive foo/
+    # file: foo/
+    ceph.dir.casesensitive="0"
+
+    $ getfattr -n ceph.dir.charmap foo/
+    # file: foo/
+    ceph.dir.charmap="{\"casesensitive\":false,\"normalization\":\"nfd\",\"encoding\":\"utf8\"}"
+
+Observe that setting the case sensitivity on a directory will cause the default
+normalization to be selected.
+
+.. note:: The normalization is applied before case folding. The directory entry name used
+          by the MDS is the case-folded and normalized name.
+
+
+Removing Character Mapping
+--------------------------
+
+If a directory is empty and not part of a snapshot, the **charmap** can be
+removed:
+
+::
+
+   $ setfattr -x ceph.dir.charmap foo/
+
+Then you can confirm it restored the normal CephFS behavior:
+
+::
+
+   $ getfattr -n ceph.dir.charmap foo/
+   foo/: ceph.dir.charmap: No such attribute
+
+If the attribute does not exist, then there is no character mapping for the
+directory. Note that a (future) child or parent directory may have a charmap
+configuration but it will have no effect on this directory. A charmap
+configuration is only inherited at directory creation.
+
+
+.. note:: The default charmap includes normalization that cannot be disabled.
```

Comment:
suggest backticks ``charmap`` inc multiple instances below.

Reply:
done in next push